src/main/java/io/cdap/plugin/couchbase/CouchbaseConfig.java

```diff
@@ -0,0 +1,340 @@
+  @Nullable
+  private String user;
+
+  @Name(CouchbaseConstants.PASSWORD)
+  @Description("Password to use to connect to the Couchbase.")
+  @Macro
+  @Nullable
+  private String password;
+
+  @Name(CouchbaseConstants.ON_ERROR)
+  @Description("Specifies how to handle error in record processing. Error will be thrown if failed to parse value " +
+    "according to provided schema.")
+  @Macro
+  private String onError;
+
+  @Name(CouchbaseConstants.SCHEMA)
+  @Description("Schema of records output by the source.")
+  private String schema;
+
+  @Name(CouchbaseConstants.MAX_PARALLELISM)
+  @Description("Maximum number of CPU cores can be used to process a query. If the specified value is less than zero" +
+    " or greater than the total number of cores in a cluster, the system will use all available cores in the cluster.")
+  private Integer maxParallelism;
+
+  @Name(CouchbaseConstants.SCAN_CONSISTENCY)
+  @Description("Specifies the consistency guarantee or constraint for index scanning.")
+  private String consistency;
+
+  @Name(CouchbaseConstants.QUERY_TIMEOUT)
+  @Description("Number of seconds to wait before a timeout has occurred on a query.")
+  private Integer timeout;
+
+  public CouchbaseConfig(String referenceName, String nodes, String bucket, String query, String user,
+                         String password, String onError, String schema, Integer maxParallelism, String consistency,
+                         Integer timeout) {
+    this.referenceName = referenceName;
+    this.nodes = nodes;
+    this.bucket = bucket;
+    this.query = query;
+    this.user = user;
+    this.password = password;
+    this.onError = onError;
+    this.schema = schema;
+    this.maxParallelism = maxParallelism;
+    this.consistency = consistency;
+    this.timeout = timeout;
+  }
+
+  public String getReferenceName() {
+    return referenceName;
+  }
+
+  public String getNodes() {
+    return nodes;
+  }
+
+  public String getBucket() {
+    return bucket;
+  }
+
+  public String getQuery() {
+    return query;
+  }
+
+  @Nullable
+  public String getUser() {
+    return user;
+  }
+
+  @Nullable
+  public String getPassword() {
+    return password;
+  }
+
+  public String getOnError() {
+    return onError;
+  }
+
+  public String getSchema() {
+    return schema;
+  }
+
+  public Integer getMaxParallelism() {
+    return maxParallelism;
+  }
+
+  public String getConsistency() {
+    return consistency;
+  }
+
+  public Integer getTimeout() {
+    return timeout;
+  }
+
+  /**
+   * Parses the json representation into a schema object.
+   *
+   * @return parsed schema object of json representation.
+   * @throws RuntimeException if there was an exception parsing the schema.
+   */
+  @Nullable
+  public Schema getParsedSchema() {
+    try {
+      return schema == null ? null : Schema.parseJson(schema);
+    } catch (IOException e) {
+      // this should not happen, since schema string comes from UI
+      throw Throwables.propagate(e);
```

Comment:
I think it should throw IllegalStateException

Reply:
Changed to throw `IllegalStateException`.